src/viam/sdk/tests/test_motion.cpp

```diff
@@ -0,0 +1,211 @@
+#define BOOST_TEST_MODULE test module test_motion
+
+#include <boost/test/included/unit_test.hpp>
+
+#include <viam/sdk/common/pose.hpp>
+#include <viam/sdk/common/world_state.hpp>
+#include <viam/sdk/services/motion/client.hpp>
+#include <viam/sdk/services/motion/motion.hpp>
+#include <viam/sdk/services/motion/server.hpp>
+#include <viam/sdk/spatialmath/geometry.hpp>
+#include <viam/sdk/tests/mocks/mock_motion.hpp>
+#include <viam/sdk/tests/test_utils.hpp>
+
+BOOST_TEST_DONT_PRINT_LOG_VALUE(viam::sdk::WorldState)
+BOOST_TEST_DONT_PRINT_LOG_VALUE(std::vector<viam::sdk::geo_obstacle>)
+
+namespace viam {
+namespace sdktests {
+namespace motion {
+
+using namespace viam::sdk;
+
+WorldState mock_world_state() {
+    WorldState::transform transform;
+    transform.reference_frame = "fake-reference-frame";
+    transform.pose_in_observer_frame = pose_in_frame("fake", {{2, 3, 4}, {5, 6, 7}, 8});
+    WorldState::geometries_in_frame obstacle;
+    obstacle.reference_frame = "ref";
+
+    return WorldState({obstacle}, {transform});
+}
+
+BOOST_AUTO_TEST_SUITE(test_mock)
+
+BOOST_AUTO_TEST_CASE(mock_move_and_get_pose) {
+    std::shared_ptr<MockMotion> motion = MockMotion::get_mock_motion();
+
+    BOOST_CHECK_EQUAL(motion->current_location, std::move(init_fake_pose()));
+
+    Name fake_name({"acme", "service", "motion"}, "fake-remote", "fake-motion");
+    auto ws = std::make_shared<WorldState>(mock_world_state());
+
+    bool success = motion->move(fake_pose(), fake_name, ws, 0, fake_map());
+
+    BOOST_TEST(success);
+    BOOST_CHECK_EQUAL(motion->current_location, fake_pose());
+    BOOST_CHECK_EQUAL(motion->peek_component_name, std::move(fake_name));
+    BOOST_CHECK_EQUAL(*(motion->peek_world_state), *ws);
+}
+
+BOOST_AUTO_TEST_CASE(mock_move_on_map) {
+    std::shared_ptr<MockMotion> motion = MockMotion::get_mock_motion();
+
+    BOOST_CHECK_EQUAL(motion->current_location, std::move(init_fake_pose()));
+
+    pose new_destination({{3, 4, 5}, {6, 7, 8}, 9});
+
+    bool success =
+        motion->move_on_map(new_destination, fake_component_name(), fake_slam_name(), fake_map());
+
+    BOOST_TEST(success);
+    BOOST_CHECK_EQUAL(motion->peek_current_pose, std::move(new_destination));
+    BOOST_CHECK_EQUAL(motion->peek_component_name, fake_component_name());
+    BOOST_CHECK_EQUAL(motion->peek_slam_name, fake_slam_name());
+}
+
+BOOST_AUTO_TEST_CASE(mock_move_on_globe) {
+    BOOST_TEST(true);
+    // std::shared_ptr<MockMotion> motion = MockMotion::get_mock_motion();
```

Comment:
Should this test be uncommented?

Reply:
:face_palm: yes it definitely should. Thanks for catching! will push up the fix.